modules/dataControllerModule/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * This module sets filters the data(userEIDs and SDA) based on the configuration.
+ * @module modules/dataController
+ */
+
+/**
+ * @interface dataControllerModule
+ */
+
+/**
+ * @function?
+ * @summary return filtered EIDs
+ * @name dataControllerModule#filterEIDs
+ * @param {Object} bidRequest
+ */
+
+/**
+ * @function?
+ * @summary return filtered SDA
+ * @name dataControllerModule#filterSDA
+ * @param {Object} bidRequest
+ */
+
+import {config} from '../../src/config.js';
+import {module, getHook} from '../../src/hook.js';
+import {processBidderRequests} from '../../src/adapters/bidderFactory.js';
+import {logError} from '../../src/utils.js';
+
+let submodules = [];
+const MODULE_NAME = 'dataController';
+let dataControllerConfig;
+
+/**
+ * enable submodule in data Controller Module
+ * @param {dataController} submodule
+ */
+export function registerSubmodules(submodule) {
+  submodules.push(submodule);
+}
+
+export function init(bidderRequests) {
+  submodules.forEach(submodule => {
+    filterData(submodule, bidderRequests)
+  });
+}
+
+export function filterData(submodule, bidRequest) {
+  dataControllerConfig = config.getConfig(MODULE_NAME);
+
+  if (!dataControllerConfig) {
+    processBidderRequests.getHooks({hook: processBidderRequestHook}).remove();
```

Comment:
I'd prefer to keep configuration and business logic separate - IMO this will get confusing fast as complexity increases. (IMO it makes more sense to turn the module off from the same place you turn it on).